### PR TITLE
fix: added failure exit code to nodes

### LINKED ISSUE #45: nodes always exit with status 0 even in error case
all ROS nodes in this package currently do not follow the standard that a success should only be returned if there actually was a success since they always do it:
https://github.com/FraunhoferIOSB/camera_aravis2/blob/8acd8251f9c77784987dc6f1abfc64bf58576e3f/camera_aravis2/src/camera_driver_uv_main.cpp#L45
https://github.com/FraunhoferIOSB/camera_aravis2/blob/8acd8251f9c77784987dc6f1abfc64bf58576e3f/camera_aravis2/src/camera_driver_gv_main.cpp#L45
https://github.com/FraunhoferIOSB/camera_aravis2/blob/8acd8251f9c77784987dc6f1abfc64bf58576e3f/camera_aravis2/src/camera_xml_exporter_main.cpp#L41
https://github.com/FraunhoferIOSB/camera_aravis2/blob/8acd8251f9c77784987dc6f1abfc64bf58576e3f/camera_aravis2/src/camera_finder_main.cpp#L45

as per [`sysexits.h`](https://www.man7.org/linux/man-pages/man3/sysexits.h.3head.html) and the [Advanced Bash Script Guide](https://tldp.org/LDP/abs/html/exitcodes.html) there are some reserved exit codes for errors (besides `0` for success). while it's not necessary to pick a special code from there it'd be good if at least exit code `1` would be used in case of errors.

the advantage of having a correct error code is that it's clearly visible from the status code that the node failed and did not complete successfully. this can then be used by scripts to e.g. automatically restart the node in an error case.

### PR DESCRIPTION
**Pull request by boitumeloruf**
This resolves #45

### CODE REVIEW

File: camera_aravis2/include/camera_aravis2/camera_xml_exporter.h [hunk 1]
```diff
@@ -60,20 +60,20 @@ class CameraXmlExporter : public CameraAravisNodeBase
...
-     * @brief Set the up launch parameters.
-     */
-    void setupParameters() override;
-
     /**
      * @brief Export XML data from camera and write to file.
      *
      * @return True if successful. False, otherwise.
      */
     [[nodiscard]] bool export_xml_data_to_file();
```
Comment (rursprung) [inline camera_aravis2/include/camera_aravis2/camera_xml_exporter.h:68]: completely unrelated, but is it intentional that you mix snake_case (this method) with camelCase (all other methods)?

File: camera_aravis2/src/camera_driver_uv_main.cpp [hunk 2]
```diff
@@ -35,12 +35,20 @@
...
+        rclcpp::spin(p_node->get_node_base_interface());
     }
+
+    //--- check if driver was initialized, i.e. if it has successfully spawned a camera stream
+    bool is_successful = p_node->isInitialized();
+
     rclcpp::shutdown();
 
-    return EXIT_SUCCESS;
+    if (is_successful)
```
Comment (rursprung) [inline camera_aravis2/src/camera_driver_uv_main.cpp:50]: you _could_ simplify this to:
```c++
return is_successful ? EXIT_SUCCESS : EXIT_FAILURE;
```

it _might_ be easier to read, but that's up to personal preference.

**Review by rursprung — APPROVED** [review]
from what i can see (i'm no expert in this code base!): LGTM 👍

File: camera_aravis2/include/camera_aravis2/camera_xml_exporter.h (hunk 1 above)
Comment (boitumeloruf) [reply camera_aravis2/include/camera_aravis2/camera_xml_exporter.h:68]: No, that is not intended. Thank you for point that out.

**Pull request merged**